spec/requests/users_spec.rb

```diff
@@ -616,18 +616,16 @@
 
   context "Revoking Sessions" do
     %w[cache sql memory].each do |session_store|
-      before do
-        ::Settings.server.session_store = session_store
-      end
+      before { TokenStore.token_caches.clear }
+      after  { TokenStore.token_caches.clear }
 
       it "revokes all own sessions of authenticated user with #{session_store}" do
+        stub_settings_merge(:server => {:session_store => session_store}) # Don't stub in before
```

Comment:
The comment here doesn't make sense out of this PR's context, so I don't think it's necessary.

Reply:
Also you should be able to stub in the before as long as you use `stub_settings_merge`

Reply:
> Also you should be able to stub in the before as long as you use stub_settings_merge

No, I couldn't, because it just runs the `memory` test 3 times when I did, instead of running each one.  See the commit message for more details

> [user_spec.rb] Stub settings on per test basis
> 
> RSpec does something weird with the `before` block here where it will run the block times per test invocation, once for each value in the array that defines `session_store`.  Each time, it would set it as `memory` instead of running each set of tests for each store respectively (determined via some "`puts` debugging")
> 
> Stubbing in the tests blocks themselves (`it` blocks), so for the sake of the tests doing what they should instead of repeating the same set of tests 3 times, the "less DRY" approach was used.

If I did the following:

```ruby
before do
  puts "before #{::Settings.server.session_store.inspect}..."
  stub_settings_merge(:server => {:session_store => session_store})
  puts "before #{::Settings.server.session_store.inspect}..."
end

it "revokes all own sessions of authenticated user with #{session_store}" do
  puts "in test #{::Settings.server.session_store.inspect}..."
```

The output for a **single test** would look like:

```
before "memory"
after "cache"
before "cache"
after "sql"
before "sql"
after "memory"
in test "memory"
```

So every test would run with `"memory"`.

Reply:
Also, more proof that this was actually an issue is that this deprecation warning:

```
.......DEPRECATION: :dalli_store will be removed in Dalli 3.0.
Please use Rails' official :mem_cache_store instead.
https://guides.rubyonrails.org/caching_with_rails.html
F.......
```

found in the build:

https://app.travis-ci.com/github/ManageIQ/manageiq-api/jobs/524811951#L2085-L2086

Wasn't there before because we always used `memory` for these specs.

Reply:
FYI, I addressed that here:

https://github.com/ManageIQ/manageiq/pull/21335

Reply:
OH i see what the problem is...because the before block is inside of the loop, it effectively creates 3 before blocks, and the most recent one is going to win.  There shouldn't be a before/after block inside of the loop even after your changes here.

Even so, I feel there has go to be a way to dry this up, but I'm not going to hold up a merge for that.

Reply:
Alright, fixed this so it is DRY.  Just had to use a `let` for the value that was set so we could pass it to the outer `before`.

Put a comment in there so people don't change it without thinking.